pkg/route/registry/route/strategy.go

```diff
@@ -43,6 +43,8 @@ func (routeStrategy) NamespaceScoped() bool {
 func (s routeStrategy) PrepareForCreate(obj runtime.Object) {
 	route := obj.(*api.Route)
 	route.Status = api.RouteStatus{}
+	// Do not allow cross-namespace routes
+	route.Spec.To.Namespace = ""
```

Comment:
LocalObjectReference doesn't have a kind, which I assume we'd want to be able to set eventually... could make it one for now, and expand later?